src/pages/Assets/Asset.tsx

```diff
@@ -53,11 +54,14 @@ export const initMarketData: MarketData = {
 }
 
 export const useAsset = () => {
+  const dispatch = useAppDispatch()
+
   const { chain, tokenId } = useParams<MatchParams>()
   const network = NetworkTypes.MAINNET
   const contractType = ContractTypes.ERC20
   const extra = tokenId ? { contractType, tokenId } : undefined
   const assetCAIP19 = caip19.toCAIP19({ chain, network, ...extra })
+  dispatch(marketApi.endpoints.findByCaip19.initiate(assetCAIP19))
```

Comment:
Can you add a comment explaining why this is here? I assume it works but I don't understand why.

Reply:
Good call, done.